components/lib/hooks/useWeb3.tsx

```diff
@@ -57,28 +58,41 @@ export const UseWeb3Provider: React.FC<ChildrenShim> = ({ children }) => {
     }
   }, []);
 
+  const idToHexString = (id: number) => {
+    return "0x" + id.toString(16);
+  };
+
   async function connectMetamask() {
     if (metamaskInstalled) {
       const newProvider = new ethers.providers.Web3Provider(window.ethereum);
-      setWeb3State({ ...web3State, connecting: true });
-      const addresses = (await newProvider.send("eth_requestAccounts", [])) as string[];
-      if (addresses.length > 0) {
-        console.log("Connected wallet ", addresses[0]);
-        const newWalletAddress = addresses[0];
-        const newSigner = newProvider.getSigner(newWalletAddress);
-        setStoredWallet(newWalletAddress);
-        setStoredProviderMode("metamask");
-        setWeb3State({
-          ...web3State,
-          connecting: false,
-          providerMode: "metamask",
-          provider: newProvider,
-          walletAddress: newWalletAddress,
-          signer: newSigner,
-        });
+
+      const chainId = await newProvider.getNetwork().then((network) => network.chainId);
+      if (chainId === Chain_Id) {
+        setWeb3State({ ...web3State, connecting: true });
+        const addresses = (await newProvider.send("eth_requestAccounts", [])) as string[];
+        if (addresses.length > 0) {
+          console.log("Connected wallet ", addresses[0]);
```

Comment:
lets just remove console.log and alert(Line# 92).

Reply:
Hi, I removed console.log and alert();